src/sentry/sentry_apps/tasks/sentry_apps.py

```diff
@@ -421,22 +421,32 @@ def workflow_notification(
 def build_comment_webhook(
     installation_id: int, issue_id: int, type: str, user_id: int, *args: Any, **kwargs: Any
 ) -> None:
-    webhook_data = get_webhook_data(installation_id, issue_id, user_id)
-    install, _, user = webhook_data
-    data = kwargs.get("data", {})
-    project_slug = data.get("project_slug")
-    comment_id = data.get("comment_id")
-    payload = {
-        "comment_id": data.get("comment_id"),
-        "issue_id": issue_id,
-        "project_slug": data.get("project_slug"),
-        "timestamp": data.get("timestamp"),
-        "comment": data.get("comment"),
-    }
-    send_webhooks(installation=install, event=type, data=payload, actor=user)
-    # `type` is comment.created, comment.updated, or comment.deleted
+    try:
+        event = SentryAppEventType(type)
+    except ValueError as e:
+        raise SentryAppSentryError(message=SentryAppWebhookFailureReason.INVALID_EVENT) from e
+
+    with SentryAppInteractionEvent(
```

Comment:
Can we avoid this by strongly typing the `type` here and forcing callers to provide us an enum instead?

Reply:
Talked IRL -> conclusion: practically yes we know that the type/event that we input into this task will be correct, so let's remove the try except and have the uncaught exception bubble up and add `ValueError` to our ignore_and_capture list since `ValueError` will only be raised when we do these "event validation checks" (I need to verify that last assumption)